README.md

```diff
@@ -1,7 +1,214 @@
-#TBA...
+# ApiProvider
 
-## Features and bugs
+### Clean and boilerplate free network requests with only a few lines of code.
 
-Please file feature requests and bugs at the [issue tracker][tracker].
+```dart
+  final request = Request(url: Path("https://someapi/user"), httpMethod: HttpMethod.GET);
 
-[tracker]: https://github.com/tikkrapp/api_provider/issues
+  final response = await apiProvider.request<User>(request: request);
+  print("fetched user :${response.body.name}");
+```
+
+### Features:
+
+- Typesafe interface on `Request` and `Response` objects.
+- Powerful `Interceptors`
+- Testing / Mocking out of the Box!
```

Comment:
Box -> box

Reply:
And decide whether to use `.`, `!`,  or nothing.

Reply:
now i am not using any punctuation marks. Looks cleaner for bullet points :)